src/util/misc.js

```diff
@@ -538,104 +515,56 @@
-     * @param {Object} [context] assign enlived props to this object (pass null to skip this)
-     * @param {(objects:fabric.Object[]) => void} callback
-     */
-    enlivenObjectEnlivables: function (object, context, callback) {
-      var enlivenProps = fabric.Object.ENLIVEN_PROPS.filter(function (key) { return !!object[key]; });
-      fabric.util.enlivenObjects(enlivenProps.map(function (key) { return object[key]; }), function (enlivedProps) {
-        var objects = {};
-        enlivenProps.forEach(function (key, index) {
-          objects[key] = enlivedProps[index];
-          context && (context[key] = enlivedProps[index]);
-        });
-        callback && callback(objects);
-      });
-    },
-
-    /**
-     * Create and wait for loading of patterns
-     * @static
-     * @memberOf fabric.util
-     * @param {Array} patterns Objects to enliven
-     * @param {Function} callback Callback to invoke when all objects are created
-     * called after each fabric object created.
+     * @param {Object} object with properties to enlive ( fill, stroke, clipPath, path )
+     * @returns {Promise<object>} the input object with enlived values
      */
-    enlivenPatterns: function(patterns, callback) {
-      patterns = patterns || [];
 
-      function onLoaded() {
-        if (++numLoadedPatterns === numPatterns) {
-          callback && callback(enlivenedPatterns);
+    enlivenObjectEnlivables: function (serializedObject) {
+      // enlive gradients or patterns
+      var enliven = {};
+      ['fill', 'stroke', 'backgroundColor', 'overlayColor', 'path', 'clipPath', 'backgroundImage', 'overlayImage'].forEach(function(prop) {
+        var value = serializedObject[prop];
+        if (!value) {
+          return;
         }
-      }
-
-      var enlivenedPatterns = [],
-          numLoadedPatterns = 0,
-          numPatterns = patterns.length;
-
-      if (!numPatterns) {
-        callback && callback(enlivenedPatterns);
-        return;
-      }
-
-      patterns.forEach(function (p, index) {
-        if (p && p.source) {
-          new fabric.Pattern(p, function(pattern) {
-            enlivenedPatterns[index] = pattern;
-            onLoaded();
-          });
+        if (value.colorStops) {
+          enliven[prop] = new fabric.Gradient(value);
+          return;
         }
-        else {
-          enlivenedPatterns[index] = p;
-          onLoaded();
+        if (value.type) {
+          enliven[prop] = fabric.util.enlivenObjects([value]);
+          return
         }
+        if (value.source) {
+          enliven[prop] = fabric.Pattern.fromObject(value);
+          return;
+        }
+        enliven[prop] = value;
+      });
+      var promises = Object.values(enliven);
+      var keys = Object.keys(enliven);
+      return Promise.all(promises).then(function(enlived) {
+        return keys.reduce(function(acc, element, index) {
+          acc[keys[index]] = enlived[index];
+          return acc;
+        }, {});
```

Comment:
@ShaMan123  i changed your work on enlivenObjectEnlivables.
I wanted it to take care of patterns/gradients/
Can you tell me what you think?

Reply:
We can add `objects` as well

Reply:
ok we find a way to do that